servers/rendering/renderer_rd/storage_rd/mesh_storage.cpp

```diff
@@ -1439,12 +1439,10 @@ void MeshStorage::multimesh_allocate_data(RID p_multimesh, int p_instances, RS::
 	multimesh->motion_vectors_current_offset = 0;
 	multimesh->motion_vectors_previous_offset = 0;
 	multimesh->motion_vectors_last_change = -1;
+	multimesh->motion_vectors_enabled = false;
```

Comment:
`multimesh_allocate_data()` can be called on an existing multimesh when it needs to be resized. This will result in the `motion_vectors_enabled` flag getting lost and having to re-allocate the buffer twice (once here and once when motion vectors are re-enabled). Is that intentional?

Also, is this change necessary for fixing the bug? To me it seems like it may have been changed as a sort of code cleanup

Reply:
Now that you point it out, I think it might not actually be necessary. OP added that at first since it did lead to some fixes, but I think the true fix might indeed just be the double buffer data copy. It's probably worth a shot to just attempt to fix it with the latter part of the fix, which was critical, the filling out both halves of the buffer.

Reply:
I think it’s necessary otherwise if you reset the size or format of an existing multimesh, the flag will stay true and the moment you set the buffer again the issue will recur (because the fix above wont do the double fill). We could probably do something to flag that it needs to be double filled in allocate data but it’s a little more complex.

Reply:
Ah, that makes sense.